.github/workflows/build-pullrequest.yml

```diff
@@ -11,6 +11,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         with:
+          fetch-depth: 0
```

Comment:
Setting fetch-depth to 0 seems to be the only major change to the GH Actions .yml files; what does it do in this case? I don't have the slightest clue about GH Actions, but it just might be good to have the rationale noted in this PR.

Reply:
GitHub does a bare checkout by default, which seem to have no enough information to make the difference check (If I understood correctly what @nedtwigg said diffplug/spotless#710)

Reply:
Correct. If you want to only format the files which have changed in the PR, then you need a git clone with enough history to determine which files changed. A bare checkout (the default on most CI platforms) doesn't have that info, but you get all the info you need by switching to `fetch-depth: 0`.

Reply:
Good to know! That explains it.